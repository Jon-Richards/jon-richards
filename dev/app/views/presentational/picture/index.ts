export { Picture, PictureProps } from './picture';
