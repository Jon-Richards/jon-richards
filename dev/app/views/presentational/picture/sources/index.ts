export { Sources, SourcesProps } from './sources';
