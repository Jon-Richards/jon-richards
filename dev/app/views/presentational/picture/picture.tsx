/** @jsx jsx */

import { css, jsx } from '@emotion/core';
import { Sources, SourcesProps } from './sources';

/**
 * Renders a Picture that supports multiple sources based on the size of the
 * viewport.
 */
export function Picture(props: PictureProps): JSX.Element {
  const { defaultSource, sources, altText } = props;

  return (
    <picture>
      <Sources sources={sources} />
      <img
        css={css`
          display: block;
          width: 100%;
          height: auto;
        `}
        src={defaultSource}
        alt={altText}
      />
    </picture>
  );
}


export interface PictureProps {
  /**
   * Alternative sources for the image to load when various media queries
   * return true.  Media queries will be evaluated based on the order in which
   * they appear in the array.  In browser, the first query to return true takes
   * precedence.
   */
  sources: SourcesProps['sources'];
  /**
   * The default source for the image.  This is used if none of the source
   * media queries evaluate to true.
   */
  defaultSource: string;
  /** The image's alt text. */
  altText: string;
}
