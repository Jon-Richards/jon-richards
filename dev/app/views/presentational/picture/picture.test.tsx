import * as React from 'react';
import { Picture, PictureProps } from './picture';
import { render } from 'enzyme';

describe('The Sources component.', () => {
  it('Should match the snapshot.', () => {
    const sources: PictureProps['sources'] = [
      { mediaQuery: 'min-width(320px)', source: 'test_small.jpg' },
      { mediaQuery: 'min-width(480px)', source: 'test_medium.jpg' },
      { mediaQuery: 'min-width(720px)', source: 'test_large.jpg' },
    ];
    const test_1 = render(
      <Picture
        defaultSource="test_medium.jpg"
        sources={sources}
        altText="A picture."
      />
    );

    expect(test_1).toMatchSnapshot();
  });

  it('Should render a source tag for each source provided.', () => {
    const sources: PictureProps['sources'] = [
      { mediaQuery: 'min-width(320px)', source: 'test_small.jpg' },
      { mediaQuery: 'min-width(480px)', source: 'test_medium.jpg' },
      { mediaQuery: 'min-width(720px)', source: 'test_large.jpg' },
    ];
    const test_1 = render(
      <Picture
        defaultSource="test_medium.jpg"
        sources={sources}
        altText="A picture."
      />
    );

    expect(test_1.find('[data-testid="picture__source"]').length).toBe(3);
  });
});
