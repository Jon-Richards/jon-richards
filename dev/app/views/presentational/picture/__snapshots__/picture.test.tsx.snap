// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Sources component. Should match the snapshot. 1`] = `
.emotion-0 {
  display: block;
  width: 100%;
  height: auto;
}

<picture>
  <source
    data-testid="picture__source"
    media="min-width(320px)"
    srcset="test_small.jpg"
  />
  <source
    data-testid="picture__source"
    media="min-width(480px)"
    srcset="test_medium.jpg"
  />
  <source
    data-testid="picture__source"
    media="min-width(720px)"
    srcset="test_large.jpg"
  />
  <img
    alt="A picture."
    class="emotion-0"
    src="test_medium.jpg"
  />
</picture>
`;
